Automattic/wp-calypso
# Framework: Add post types Redux actions and reducer

This pull request seeks to add post types to the Redux global state tree, including actions and a reducer for managing the request and storage of post types for a given site.

This is planned to be used in the `<PostSelector />` component for displaying a label associated with the post type of each rendered option when a `type` prop of `any` is specified for the selector (related: #3197).

__Implementation notes:__

@gwwar : I found that `tv4.validateResult` _throws an error_ when the result is invalid, so our logic for warning was never being reached. Is this expected?

__Testing instructions:__

There are no visual effects of these changes.

Verify that the inclusion of `postTypes` in the global Redux state tree does not prevent Calypso from loading.

Ensure that Mocha tests pass by running `make test` in the project root directory or directly from `client/state/post-types`.
